### ISSUE TITLE
stamp issuer bucket data should be persisted more often during uploads

### ISSUE THREAD

**Issue by istae** [ethersphere/bee#4544]
as part of the upload process, before chunks are sent out to the network, each chunk is stamped using the stamp issuer.
each stamp contains a specific bucket index that is incremented for every chunk that lands in that particular bucket.

Side note: There are by default 2^16 buckets per batch. The max hits per bucket is defined by the batch's depth.

The bucket and the data are kept in memory and are only persisted when the upload finishes. If for whatever reason the node shutdown in the middle of an upload, the newly assigned bucket indexes are not persisted to disk. This can cause uploads in the future to reuse those indices. In the case of mutable batches, chunks that were previously assigned to those indices will be replaced with the new chunks, causing data loss, and in the case of immutable batches, the new chunks will be rejected.

In short, it's important that the bucket data is persisted more often. How often have to be decided with caution as the bucket data as it is today is a few hundred kilobytes.

**Comment by istae** [comment]
closing this one as this PR https://github.com/ethersphere/bee/pull/4559 has changed the requirements.
